File: medical-appointment-booking/02-backend/medical-system/src/main/java/com/johnabbott/medical_system/entity/Appointment.java
package com.johnabbott.medical_system.entity;

import lombok.Data;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.Date;
import jakarta.persistence.*;

@Entity
@Table(name = "appointment")
@Data
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String patientName;

    @Column(nullable = false)
    private Long patientId;

    @Column(nullable = false)
    private String doctorName;

    @Column(nullable = false)
    private Long doctorId;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false)
    private LocalTime time;


    private String description;

    @Column(nullable = false)
    private boolean status;

}

File: medical-appointment-booking/03-frontend/react-medical-booking/src/Models/Appointment.ts
export interface Appointment {
  id?: number;
  patientName: string;
  patientId: number;
  doctorName: string;
  doctorId: number;
  date: string; // ISO date string
  time: string; // Time in HH:mm format
  description: string;
  status: boolean;
}